forge-game/src/main/java/forge/game/ability/effects/ChooseTypeEffect.java

```diff
@@ -154,4 +154,12 @@ public void resolve(SpellAbility sa) {
             throw new InvalidParameterException(sa.getHostCard() + "'s ability resulted in no types to choose from");
         }
     }
+
+    private static List<String> getInvalidTypes(SpellAbility sa) {
+        if(sa.getHostCard().hasAnyNotedType())
```

Comment:
this check is already done in in line 119

it is also important to check for hasParam "Note"

Reply:
I was getting `empty`, which didn't sit well with me, but It was meant to be the first time selection.

Pushing an update now